common.zig

```diff
@@ -4,6 +4,17 @@ const common = @This();
 
 pub const SocketReader = std.io.Reader(std.posix.socket_t, std.posix.RecvFromError, readSocket);
 
+/// Sanity check that we're not running into data integrity (corruption) issues caused
+/// by overflowing and wrapping around to the front ofq the buffer.
+fn checkMessageLengthFitsInBuffer(message_length: usize, buffer_limit: usize) !void {
```

Comment:
Re: `checkMessageLengthFitsInBuffer` usage, this already has a discussion on https://github.com/marler8997/zigx/pull/11#discussion_r1342155666